tools/backups2datalad/aioutil.py

```diff
@@ -220,14 +220,14 @@ async def stream_null_command(
         await anyio.open_process(argstrs, cwd=cwd, stderr=None), desc
     ) as p:
         assert p.stdout is not None
-        async with TextReceiveStream(p.stdout) as stream:
-            try:
-                splitter = TerminatedSplitter("\0", retain=False)
-                async for chunk in splitter.aitersplit(stream):
-                    yield chunk
-            except BaseException:
-                log.exception("Exception raised while handling output from %s", desc)
-                raise
+        try:
+            stream = TextReceiveStream(p.stdout)
+            splitter = TerminatedSplitter("\0", retain=False)
+            async for chunk in splitter.aitersplit(stream):
+                yield chunk
+        except BaseException:
+            log.exception("Exception raised while handling output from %s", desc)
+            raise
```

Comment:
does stream need its `await stream.aclose()` in `finally:` or at the end of the try block?

Reply:
I don't think it really needs to be called at all.  `stream.aclose()` just closes the underlying stream, which is here `p.stdout`, but that should be taken care of by the `open_process()` context manager.